content/en/docs/concepts/workloads/_index.md

```diff
@@ -8,31 +8,53 @@ no_list: true
-that make sure the right number of the right kind of Pod are running, to match the state
+that make sure the right number of the right kind of `Pod` are running, to match the state
 you specified.
 
-Those workload resources include:
+Kubernetes provides several built-in workload resources:
 
-* [Deployment](/docs/concepts/workloads/controllers/deployment/) and [ReplicaSet](/docs/concepts/workloads/controllers/replicaset/)
-  (replacing the legacy resource {{< glossary_tooltip text="ReplicationController" term_id="replication-controller" >}});
-* [StatefulSet](/docs/concepts/workloads/controllers/statefulset/);
-* [DaemonSet](/docs/concepts/workloads/controllers/daemonset/) for running Pods that provide
-  node-local facilities, such as a storage driver or network plugin;
-* [Job](/docs/concepts/workloads/controllers/job/) and
-  [CronJob](/docs/concepts/workloads/controllers/cron-jobs/)
-  for tasks that run to completion.
+* [`Deployment`](/docs/concepts/workloads/controllers/deployment/) and [`ReplicaSet`](/docs/concepts/workloads/controllers/replicaset/)
+  (replacing the legacy resource
+  {{< glossary_tooltip text="ReplicationController" term_id="replication-controller" >}}).
+  `Deployment` is a good fit for managing a stateless application workload on your cluster,
+  where any `Pod` in the `Deployment` is interchangeable and can be replaced if needed.
+* [`StatefulSet`](/docs/concepts/workloads/controllers/statefulset/) lets you
+  run one or more related Pods that do track state somehow. For example, if your workload
+  records data persistently, you can run a `StatefulSet` that matches each `Pod` with a
+  [`PersistentVolume`](/docs/concepts/storage/persistent-volumes/), and replicate both
+  the pods and their storage across different failure zones. Your code, running in the `Pods`,
+  can replicate data to other `Pods` in the same `StatefulSet` to improve overall resilience.
+* [`DaemonSet`](/docs/concepts/workloads/controllers/daemonset/) defines `Pods` that provide
+  node-local facilities. These might be fundamental to the operation of your cluster, such
+  as a networking helper tool, or be part of an
+  {{< glossary_tooltip text="add-on" term_id="addons" >}}.  
+  Every time you add a `Node` to your cluster that matches the specification in a `DaemonSet`,
+  the control plane schedules a `Pod` for that `DaemonSet` onto the new node.
+* [`Job`](/docs/concepts/workloads/controllers/job/) and
+  [`CronJob`](/docs/concepts/workloads/controllers/cron-jobs/)
+  define tasks that run to completion and then stop. Jobs represent one-off tasks, whereas
+  CronJobs recur according to a schedule.
+
+In the wider Kubernetes ecosystem, you can find third-party workload resources that provide
+additional behaviors. Using a
+[custom resource definition](/docs/concepts/extend-kubernetes/api-extension/custom-resources/),
+you can add in a third-party workload resource if you want a specific behavior that's not part
```

Comment:
Should you reference,
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#customresourcedefinition-v1-apiextensions-k8s-io

Reply:
I'd rather link to the concept than the API details, especially whilst the API details don't hyperlink back to the concept.